chore(CMSIS,PeriphDrivers): Sync latest MAX32657 changes

ozersa opened the pull request:
This commit content comes from feat/ME30 branch.
CMSIS register, includes files and peripheral source files copied from feat/ME30 branch to here to merge them into main.
These files requires to push MAX32657 development to the upstream.

### Checklist Before Requesting Review

- [X] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

ozersa committed:
chore(CMSIS,PeriphDrivers): Sync latest MAX32657 changes

This commit content comes from feat/ME30 branch.
CMSIS register, includes, peripheral source files copied from feat/ME30 branch to here
to these files being merged into main.
These files requires to push MAX32657 to the upstream.

Co-authored-by: Jake-Carter <Jake.Carter@analog.com>
Co-authored-by: sihyung-maxim <Sihyung.Woo@analog.com>
Co-authored-by: EricB-ADI <eric.busch@analogdevices.com>
Co-authored-by: Bobby <Bobby.Counts@analog.com>
Co-authored-by: Anıl Kara <anil.kara@analog.com>
Co-authored-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>
Co-authored-by: Sadik Ozer <sadik.ozer@analog.com>

ozersa commented:
@ttmut please check i3c_reva.c file, this file content not same for main and feat/ME30 branch,  the content  seems changed while merging  https://github.com/analogdevicesinc/msdk/pull/1111 
I keep main/i3c_reva.c file.

ttmut commented on Libraries/PeriphDrivers/Include/MAX32657/i3c.h:
```diff
@@ -153,9 +153,9 @@ struct _i3c_ccc_req_t {
 /**
- * @brief Private SDR/I2C request structure.
+ * @brief Private SDR request structure.
  *
- * SDR/I2C read and write to an I3C target.
+ * SDR read and write to an I3C target.
```
Not critical but `SDR/I2C ...` is the correct comment.

ozersa replied:
Feel free to push a new commit to fix it.

ttmut commented:
> @ttmut please check i3c_reva.c file, this file content not same for main and feat/ME30 branch, the content seems changed while merging #1111 I keep main/i3c_reva.c file.

`main/i3c_reva.c` is the correct one. Thanks.

ozersa committed:
run clang-format

Signed-off-by: Sadik Ozer <sadik.ozer@analog.com>

ttmut committed:
chore(PeriphDrivers): Minor fix in I3C API documentation

mxc_i3c_req_t is used for both SDR and I2C mode transfers. Reflect this
in structure description.

Signed-off-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>

Jake-Carter reviewed (commented):
Thanks @ozersa 

I copied your merge conflict resolutions back onto #1075 

Let's just merge that one in now

ozersa commented:
> Thanks @ozersa
> 
> I copied your merge conflict resolutions back onto #1075
> 
> Let's just merge that one in now

It is ok for me.
